src/bkl/plugins/vsbase.py

```diff
@@ -594,7 +594,22 @@ def write(self):
                        (prj.kind, prj.name, self.formatter.format(prj.projectfile), str(prj.guid)))
             if prj.dependencies:
                 outf.write("\tProjectSection(ProjectDependencies) = postProject\n")
+                # We may happen to have duplicate dependencies, e.g. a
+                # dependency could be defined explicitly and also inherited
+                # from a base template, but we shouldn't duplicate them in the
+                # solution file as MSVS doesn't do it and would rewrite them
+                # to get rid of the duplicates if we generated them.
+                #
+                # Note that we still want to specify them in the order they
+                # were given, which means we can't simply iterate over
+                # set(prj.dependencies) and need to use this temporary set
+                # instead.
+                already_seen=set()
```

Comment:
My first draft said then it would probably be simpler to instead write
```python
for d in OrderedSet(prj.dependencies):
    ...
```

But then I grepped the code for such uses and found something better:
```python
from bkl.utils import filter_duplicates
...
for d in filter_duplicates(prj.dependencies):
    outf.write("\t\t{%(g)s} = {%(g)s}\n" % {'g':self._get_target_guid(d)})
```
Probably doesn't even need the explanatory comments then — it says what it does and it's clear why one might want to eliminate duplicates here...

Reply:
Great, thanks for the `filter_duplicates()` hint!

I'd still keep the comment because it might not be obvious that it's more than just an "optimization".